src/box/box.cc

```diff
@@ -1805,6 +1806,7 @@ box_issue_demote(uint32_t prev_leader_id, int64_t promote_lsn)
 	assert(box_raft()->volatile_term == box_raft()->term);
 	assert(promote_lsn >= 0);
 	txn_limbo_begin(&txn_limbo);
+	box_raft_wait_term_broadcasted();
```

Comment:
While you waited for the term broadcast, the term could already be bumped by another node. And below you will write a demote for a wrong term. Same above for promote.